src/analysis/processing/qgsalgorithmroundrastervalues.cpp

```diff
@@ -0,0 +1,262 @@
+/***************************************************************************
+                         qgsalgorithmroundrastervalues.cpp
+                         ---------------------
+    begin                : April 2020
+    copyright            : (C) 2020 by Clemens Raffler
+    email                : clemens dot raffler at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmroundrastervalues.h"
+#include "qgsrasterfilewriter.h"
+
+///@cond PRIVATE
+
+QString QgsRoundRasterValuesAlgorithm::name() const
+{
+  return QStringLiteral( "roundrastervalues" );
+}
+
+QString QgsRoundRasterValuesAlgorithm::displayName() const
+{
+  return QObject::tr( "Round raster" );
+}
+
+QStringList QgsRoundRasterValuesAlgorithm::tags() const
+{
+  return QObject::tr( "data,cells,round,truncate" ).split( ',' );
+}
+
+QString QgsRoundRasterValuesAlgorithm::group() const
+{
+  return QObject::tr( "Raster tools" );
+}
+
+QString QgsRoundRasterValuesAlgorithm::groupId() const
+{
+  return QStringLiteral( "rastertools" );
+}
+
+void QgsRoundRasterValuesAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterRasterLayer( QStringLiteral( "INPUT" ), QStringLiteral( "Input raster" ) ) );
+  addParameter( new QgsProcessingParameterBand( QStringLiteral( "BAND" ), QObject::tr( "Band number" ), 1, QStringLiteral( "INPUT" ) ) );
+  addParameter( new QgsProcessingParameterEnum( QStringLiteral( "ROUNDING_DIRECTION" ), QObject::tr( "Rounding direction" ), QStringList() << QObject::tr( "Round up" ) << QObject::tr( "Round to nearest" ) << QObject::tr( "Round down" ), false, 1 ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "DECIMAL_PLACES" ), QObject::tr( "Number of decimals places (use negative values to round cell values to a multiple of a base n, see advanced parameters)" ), QgsProcessingParameterNumber::Integer, 2 ) );
```

Comment:
This description is very long compared with the usual parameter description. I'd shorten to just "number of decimal places" and leave the full explanation for the help string.

I keep thinking we need api to support extended descriptions for parameters, which we could show in a tooltip...

Reply:
Agreed - this would be very nice!

Reply:
fixed